tools/certify/commands/show-cert.cpp

```diff
@@ -0,0 +1,287 @@
+
+    for (auto& subject_attr : cert.subject_attributes) {
+        SubjectAttributeType attr_type = get_type(subject_attr);
+        if (attr_type == SubjectAttributeType::Assurance_Level) {
+            SubjectAssurance assurance = boost::get<SubjectAssurance>(subject_attr);
+            std::cout << "Assurance: " << (assurance.raw & assurance.assurance_mask);
+            std::cout << " with a confidence of " <<  (assurance.raw & assurance.confidence_mask);
+            std::cout << std::endl << std::endl;
+        } else if (attr_type == SubjectAttributeType::Its_Aid_List) {
+            std::list<IntX> its_application_ids = boost::get<std::list<IntX>>(subject_attr);
+
+            std::cout << "ITS Application IDs:" << std::endl;
+            if (its_application_ids.size() == 0) {
+                std::cout << "None";
+            } else {
+                for (auto& its_application_id : its_application_ids) {
+                    certificate_application_ids++;
+
+                    std::cout << " - ";
+                    if (its_application_id == itsAidCa) {
+                        std::cout << "36 (CA-Basic service)";
+                    } else if (its_application_id == itsAidDen) {
+                        std::cout << "37 (DEN-Basic service)";
+                    } else {
+                        std::cout << its_application_id.get();
+                    }
+                    std::cout << std::endl;
+                }
+            }
+            std::cout << std::endl;
+        } else if (attr_type == SubjectAttributeType::Its_Aid_Ssp_List) {
+            std::list<ItsAidSsp> its_service_specific_permissions = boost::get<std::list<ItsAidSsp>>(subject_attr);
+            for (auto& its_ssp : its_service_specific_permissions) {
+                if (its_ssp.its_aid == itsAidCa) {
+                    std::cout << "CA - ITS Service Specific Permissions:" << std::endl;
+                    ByteBuffer& ssp = its_ssp.service_specific_permissions;
+
+                    if (ssp.size() == 0) {
+                        std::cerr << "Invalid service specific permissions for CA" << std::endl;
+                        continue;
+                    }
+
+                    // See final draft ETSI EN 302 637-2 V1.3.1 (2014-09)
+                    if (ssp[0] == 0) {
+                        if (ssp.size() != 1) {
+                            std::cout << " - Warning: ssp.size() is expected to be 1, but was " << ssp.size() << std::endl;
+                        } else {
+                            std::cout << " - No version, shell be used only for testing." << std::endl;
+                        }
+                    } else if (ssp[0] == 1) {
+                        if (ssp.size() != 3) {
+                            std::cout << " - Warning: ssp.size() is expected to be 3, but was " << ssp.size() << std::endl;
+                        } else {
+                            if (ssp[1] & (1 << 7)) {
```

Comment:
I think all this SSP decoding belongs to vanetza::security.

Reply:
Eventually, yes, but I'm not sure which names to use for the values of an enum.